### PR TITLE
Rename rights, lefts, both

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes:https://github.com/composewell/streamly/issues/2135

File: core/src/Streamly/Internal/Data/Fold/Type.hs [hunk 1]
```diff
@@ -1290,19 +1290,19 @@ filterM f (Fold step begin done) = Fold step' begin done
...
 --
-{-# INLINE rights #-}
-rights :: (Monad m) => Fold m b c -> Fold m (Either a b) c
-rights = filter isRight . lmap (fromRight undefined)
+{-# INLINE catRights #-}
+catRights :: (Monad m) => Fold m b c -> Fold m (Either a b) c
+catRights = filter isRight . lmap (fromRight undefined)
 
--- | Remove the either wrapper and flatten both lefts and as well as rights in
+-- | Remove the either wrapper and flatten both catLefts and as well as catRights in
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Fold/Type.hs:1305]:
```suggestion
-- | Remove the either wrapper and flatten both lefts and as well as rights in
```

File: core/src/Streamly/Internal/Data/Parser/ParserD/Type.hs [hunk 2]
```diff
@@ -284,7 +284,7 @@ instance Functor (Initial s) where
 -- are returning the leftover of that. The driver may choose to restart from
 -- the last segment where this parser failed or from the beginning.
 --
--- Folds can only return the right values. Parsers can also return lefts.
+-- Folds can only return the right values. Parsers can also return catLefts.
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Parser/ParserD/Type.hs:287]:
```suggestion
-- Folds can only return the right values. Parsers can also return lefts.
```

File: core/src/Streamly/Internal/Data/Stream/Expand.hs [hunk 3]
```diff
@@ -688,7 +688,7 @@ iterateSmapMWith combine f initial stream =
 -- Either streams
 ------------------------------------------------------------------------------
 
--- Keep concating either streams as long as rights are generated, stop as soon
+-- Keep concating either streams as long as catRights are generated, stop as soon
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/Expand.hs:691]:
```suggestion
-- Keep concating either streams as long as rights are generated, stop as soon
```

File: src/Streamly/Internal/Data/Stream/IsStream/Expand.hs [hunk 4]
```diff
@@ -1111,7 +1111,7 @@ iterateSmapMWith combine f initial stream =
 -- Either streams
 ------------------------------------------------------------------------------
 
--- Keep concating either streams as long as rights are generated, stop as soon
+-- Keep concating either streams as long as catRights are generated, stop as soon
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Stream/IsStream/Expand.hs:1114]:
```suggestion
-- Keep concating either streams as long as rights are generated, stop as soon
```

File: core/src/Streamly/Internal/Data/Stream/Transform.hs [hunk 5]
```diff
@@ -998,31 +998,31 @@ mapMaybeM f = fmap fromJust . filter isJust . mapM f
...
 --
-{-# INLINE rights #-}
-rights :: Monad m => Stream m (Either a b) -> Stream m b
-rights = fmap (fromRight undefined) . filter isRight
+{-# INLINE catRights #-}
+catRights :: Monad m => Stream m (Either a b) -> Stream m b
+catRights = fmap (fromRight undefined) . filter isRight
 
--- | Remove the either wrapper and flatten both lefts and as well as rights in
+-- | Remove the either wrapper and flatten catEithers catLefts and as well as catRights in
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/Transform.hs:1019]:
```suggestion
-- | Remove the either wrapper and flatten both lefts and as well as rights in
```

**Review by harendra-kumar — APPROVED** [review]

File: src/Streamly/Internal/FileSystem/Event/Darwin.hs [hunk 6]
```diff
@@ -505,7 +505,7 @@ readOneEvent = do
 
 watchToStream :: Watch -> Stream IO Event
 watchToStream (Watch handle _ _) =
-    S.catRights $ S.parseMany readOneEvent $ S.unfold FH.reader handle
+    S.rights $ S.parseMany readOneEvent $ S.unfold FH.reader handle
```
Comment (harendra-kumar) [inline src/Streamly/Internal/FileSystem/Event/Darwin.hs:508]: We should remove the import of import qualified Streamly.Internal.Data.Stream.IsStream as S from this module as it is deprecated. and should use Stream.catRights here.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: core/src/Streamly/Internal/Data/Stream/Top.hs [hunk 7]
```diff
@@ -49,6 +49,7 @@ import Data.IORef (newIORef, readIORef, modifyIORef')
 import Data.Maybe (isJust)
 import Streamly.Internal.Data.Stream.Common ()
 import Streamly.Internal.Data.Stream.Cross (CrossStream (..))
+--import Streamly.Internal.Data.Stream (Stream, rights, fromStreamD, toStreamD)
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/Top.hs:52]:
```suggestion
```

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**